geth/node/utils.go

```diff
@@ -46,3 +52,101 @@ func HaltOnInterruptSignal(nodeManager *NodeManager) {
 	}
 	panic("interrupted!")
 }
+
+// logMessageStat logs stat derived from provided DeliveryState.
+func logMessageStat(state notifications.DeliveryState) {
+	stat := generateMessageStat(state)
+	statdata, err := json.Marshal(stat)
+	if err != nil {
+		log.Info(fmt.Sprintf("%s : JSON MARHSAL ERROR : %+q", params.MessageStatHeader, err))
+		return
+	}
+
+	log.Info(fmt.Sprintf("%s : %s : %s : %s : %+s", params.MessageStatHeader, stat.Protocol, stat.Type, stat.Status, string(statdata)))
```

Comment:
What's the reasone for not following `log.Info("Whisper message", "label1", value1, "label2", value2, ...)`?

Reply:
I can identify easily the log line i need and easily split and parse all components appropriately.